packages/version-bump/src/tests/main.test.ts

```diff
@@ -0,0 +1,70 @@
+import * as core from '@actions/core';
+import * as io from '@eventespresso-actions/io';
+import run from '../main';
+import { getMockedFileContents } from './utils';
+
+const input = {
+	'info-json-file': 'info.json',
+	'main-file': 'main-file.php',
+	'readme-file': 'readme.txt',
+	type: 'pre_release',
+};
+let output: Record<string, string> = {};
+
+const testCases = [
+	{
+		type: 'pre_release',
+		inputVer: '4.10.9.rc.011',
+		outputVer: '4.10.9.beta',
+	},
+	{
+		type: 'minor',
+		inputVer: '4.10.9.rc.011',
+		outputVer: '4.11.0.p',
+	},
+];
```

Comment:
These are just example cases. It needs plenty of test cases which can be added in separate PRs.